app/src/main/java/com/android/voyageur/ui/trip/AddActivity.kt

```diff
@@ -84,12 +79,27 @@ fun AddActivityScreen(tripsViewModel: TripsViewModel, navigationActions: Navigat
 
     val today = normalizeToMidnight(Date())
     val dateNormalized = activityDate?.let { normalizeToMidnight(Date(it)) } ?: Date(0)
+    val selectedTrip = tripsViewModel.selectedTrip.value!!
+
+    if (activityDate != null &&
+        (dateNormalized.after(normalizeToMidnight(selectedTrip.endDate.toDate())) ||
+            dateNormalized.before(normalizeToMidnight(selectedTrip.startDate.toDate())))) {
+      Toast.makeText(
+              context, "Please select a date within the trip's scheduled dates", Toast.LENGTH_SHORT)
+          .show()
+      return
+    }
 
     if (activityDate != null && dateNormalized.before(today)) {
       Toast.makeText(context, "The activity date cannot be in the past", Toast.LENGTH_SHORT).show()
       return
     }
 
+    if (startTime == null && endTime != null) {
```

Comment:
I've noticed you have a lot of if conditions like this which throw toasts, I think it would improve code clarity if you structured them with a _when_ condition.

Reply:
Good idea, I made the changes.